Native tests fail in Gradle-based Quarkus project when module has no classes in `src/main/java`

### Describe the bug

Hello,

We’ve encountered an unusual bug in our current Quarkus project setup. We are using Gradle to build the project, and one of the modules does not contain any classes in `src/main/java`. However, this module includes native tests. The native tests fail to execute when running them in this setup. Here's an example message of a failing native test:
```
MyNativeTest > test() FAILED
    java.lang.RuntimeException at QuarkusIntegrationTestExtension.java:373
        Caused by: java.lang.IllegalArgumentException at PathTree.java:53
            Caused by: java.nio.file.NoSuchFileException at UnixException.java:92
```
The `NoSuchFileException` message is `java.nio.file.NoSuchFileException: <path_removed_for_clarity>/build/classes/java/main`

As a **workaround**, when we add an empty class in `src/main/java`, the native test executes correctly.

### Expected behavior

I would expect the native tests to be executed even if the module has no source.

### Actual behavior

The native tests fail when the module has no source.

### How to Reproduce?

Reproducer:  https://github.com/celcius112/quarkus-native-test-empty-main

The steps to reproduce can be found in the project's README.


### Output of `uname -a` or `ver`

Darwin mac.lan 24.2.0 Darwin Kernel Version 24.2.0: Fri Dec  6 19:02:41 PST 2024; root:xnu-11215.61.5~2/RELEASE_ARM64_T6030 arm64

### Output of `java -version`

openjdk version "21.0.2" 2024-01-16 
OpenJDK Runtime Environment GraalVM CE 21.0.2+13.1 (build 21.0.2+13-jvmci-23.1-b30) 
OpenJDK 64-Bit Server VM GraalVM CE 21.0.2+13.1 (build 21.0.2+13-jvmci-23.1-b30, mixed mode, sharing)

### Quarkus version or git rev

3.17.8

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

8.9 (reproduced with 8.12)

### Additional information

_No response_

--- a/test-framework/junit5/src/main/java/io/quarkus/test/junit/IntegrationTestUtil.java
+++ b/test-framework/junit5/src/main/java/io/quarkus/test/junit/IntegrationTestUtil.java
@@ -193,10 +193,14 @@ public final class IntegrationTestUtil {
         runnerBuilder.setProjectRoot(projectRoot);
         runnerBuilder.setTargetDirectory(PathTestHelper.getProjectBuildDir(projectRoot, testClassLocation));
 
-        rootBuilder.add(appClassLocation);
+        if (Files.exists(appClassLocation)) {
+            rootBuilder.add(appClassLocation);
+        }
         final Path appResourcesLocation = PathTestHelper.getResourcesForClassesDirOrNull(appClassLocation, "main");
         if (appResourcesLocation != null) {
-            rootBuilder.add(appResourcesLocation);
+            if (Files.exists(appResourcesLocation)) {
+                rootBuilder.add(appResourcesLocation);
+            }
         }
 
         // If gradle project running directly with IDE

